### PR TITLE
Replace expensive query with cheaper one

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Replace expensive query with cheaper one

Before
----------------------------------------
call so `apiQuery` - retrieves all fields across several tables

After
----------------------------------------
Apiv4 field - retrieves 1 field

Technical Details
----------------------------------------
Rather than retrieve everything with a legacy function, just get what we want....

Note I commented on how silly I think it is but no change to outcome

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
well test failure is related @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
@seamuslee001 it passed in isolation - my guess is there is a 'spare' Anthony Anderson in the db :-) Will check

File: CRM/Core/BAO/MessageTemplate.php [hunk 1]
```diff
@@ -369,16 +370,17 @@ public static function sendTemplate($params) {
...
-      [$contact] = CRM_Contact_BAO_Query::apiQuery($contactParams);
-
-      $prefs = array_pop($contact);
-
-      if (isset($prefs['preferred_mail_format']) and $prefs['preferred_mail_format'] === 'HTML') {
+      // @todo - consider whether we really should be loading
+      // this based on 'the first email in the db that matches'.
+      // when we likely have the contact id. OTOH people probably barely
+      // use preferredMailFormat these days - the good fight against html
+      // emails was lost a decade ago...
```
Comment (colemanw) [inline CRM/Core/BAO/MessageTemplate.php:377]: I would fully support ripping out 'preferred_mail_format'.

Comment (eileenmcnaughton) [reply CRM/Core/BAO/MessageTemplate.php:377]: @jmcclelland given you just put up a PR about importing this field I'm thinking you might still see value in it? I am a bit confused why we wouldn't always send 'BOTH' and let the mail client decide

Comment (JoeMurray) [reply CRM/Core/BAO/MessageTemplate.php:377]: I don't think sending in non-html ever took off. I'd be in favour of obsoleting then later removing this.

**Comment by jmcclelland** [comment]
My recent [pull request](21433) was on `preferred_communication_method` (e.g. Email vs SMS vs phone call), not `preferred_mail_format`. I agree that we should yank out `preferred_mail_format` and always send both text and html in each email messages.

**Pull request merged**